test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,172 @@
+/*!
+ * Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+const confirm = jest.fn().mockResolvedValue(true);
+jest.mock('cli-ux', () => {
+  // Mocking needs to happen before the command is imported
+  const cli = jest.requireActual('cli-ux');
+  return {
+    ...cli,
+    default: {
+      ...cli.default,
+      confirm
+    }
+  };
+});
+
+const warn = jest.fn(message => console.log('MOCKED WARNING: ', message));
+jest.mock('@oclif/command', () => {
+  const command = jest.requireActual('@oclif/command');
+  command.Command.prototype.warn = warn;
+  return command;
+});
+
+import { GeneratorOptions as GeneratorOptionsSDK, generatorOptionsCli as generatorOptionsSDK } from '@sap/cloud-sdk-generator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { FlagsParsed } from '../src/commands/generate-vdm';
+import * as generateVdmUtil from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(generateVdmUtil, 'toGeneratorSDK');
+
+describe('generate-vdm', () => {
+  const pathForTests = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');
+
+  beforeAll(() => {
+    const pathForResources = path.resolve(__dirname, 'resources', 'template-generator-vdm');
+    fs.copySync(pathForResources, pathForTests);
+  });
+
+  afterAll(() => {
+    fs.removeSync(pathForTests);
+  });
+
+  it('should generate a vdm.', async () => {
+    await GenerateVdm.run(['-i', 'edmxSource', '-o', 'generatedVdm', '--forceOverwrite', '--projectDir', pathForTests]);
+    const files = fs.readdirSync(path.join(pathForTests, 'generatedVdm', 'yy-1-socialnetworkaccount-cds-service'));
+    expect(files.length).toBeGreaterThan(0);
+  }, 60000);
+
+  it('should fail if the mandatory parameters are not there', async () => {
+    try {
+      await GenerateVdm.run([]);
+    } catch (e) {
+      expect(e.message).toContain('-i, --inputDir INPUTDIR');
+    }
+  });
+
+  it('should parse strings and with --no- it should lead to all false.', async () => {
+    const expected = getAllFalse();
+    delete expected.projectDir;
```

Comment:
getAllFalse() does not contain `projectDir` as far as I can tell, why delete it?

Reply:
the getAllFalse() contains the project dir line 162.

Reply:
I don't think that `getAllFalse()` is a good name. Also I would propose to sort the properties below so that all non-boolean properties are in a block.

Reply:
changed the name